fpga_interchange/chip_info.py

```diff
@@ -632,6 +632,149 @@ def append_bba(self, bba, label_prefix):
             bba.u32(len(getattr(self, field)))
 
 
+class ChainPattern():
+    def __init__(self, source, sink):
+        self.source = source
+        self.sink = sink
+
+    def append_children_bba(self, bba, label_prefix):
+        pass
+
+    def append_bba(self, bba, label_prefix):
+        bba.str(self.source)
+        bba.str(self.sink)
+
+
+class ChainCoordConfig():
+    def __init__(self, coord, step):
+        self.coord = coord
+        self.step = step
+
+    def append_children_bba(self, bba, label_prefix):
+        pass
+
+    def append_bba(self, bba, label_prefix):
+        bba.str(self.coord)
```

Comment:
this should definitely be an enum, not an string, similar to my comment on the schema